docs: oel migrate and upgrade guide (#1804)
ory/docs: docs/self-hosted/oel/oauth2/migrate.mdx

## Prerequisites

Before starting the upgrade process, ensure you meet the following requirements:

1. **Valid Ory Enterprise License**: You must have purchased an Ory Enterprise License.
2. **Access to the Ory Enterprise Docker Registry**: You need access to download the OEL Docker images.
3. **Ory Hydra Version**: Your current Ory Hydra installation must be version 2.0 or higher.
4. **Backup and Testing**: Create a backup of your current Ory Hydra database and test the migration on a test environment to
   ensure compatibility and minimize risks.

## Upgrade process

Follow these steps to upgrade from Ory Hydra open source to Ory Hydra Enterprise License (OEL):

### 1. Keep the open source version running

Ensure that your current open source version of Ory Hydra is running.

### 2. Run the OEL Image for database migration

Use the Ory Hydra Enterprise License Docker image to run the `hydra migrate sql up` command against the database of the open
source service. This step applies the necessary database migrations for OEL.

```bash
docker run --rm \
  -e DSN=your_database_connection_string \
  europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<version> \

>   migrate sql up -e -f config.yml

Reviewer: Let's mention that the config here which is OS one is compatible with OEL and no changes are required - would be more clear for the end user, WDYT?

Author: 👍

```

Replace `your_database_connection_string` with your actual database connection string.

### 3. Switch to Ory Hydra Enterprise License

After successfully migrating the database:

> 1. **Stop the Open Source Version**: Shut down your current open source Ory Hydra service.

Reviewer: Executing those steps means there is a small downtime as the OS version goes down and oel is up after. Maybe we should also mention that in the high level overview.
To reduce the downtime window the OEL Hydra can be provisioned before the actual switch is done but just not connected to DB - this would also reduce bit a time, wdyt?

Author: Yes, do you have a suggestion how I could improve this?

Reviewer: added suggestion

> 1. **Stop the Open Source Version**: Shut down your current open source Ory Hydra service.
> 2. **Start the Ory Hydra Enterprise License Version**: Launch the Ory Hydra OEL using the same database connection string and
>    configuration, but with the OEL Docker image.
> 
> ```bash
> docker run -d \
>   -e DSN=your_database_connection_string \
>   -p 4444:4444 -p 4445:4445 \
>   europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<version> \
>   serve all -c config.yml
> ```
> 
> Ensure to replace `your_database_connection_string` with your database connection string used in the migration step.

Reviewer: ```suggestion
After successfully migrating the database:

1. **Start the Ory Hydra Enterprise License Version**: Launch the Ory Hydra OEL using the same database connection string and configuration, but with the OEL Docker image.

```bash
docker run -d \
  --name=hydra-oel \
  -e DSN=your_database_connection_string \
  -p 4444:4444 -p 4445:4445 \
  europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<version> \
  serve all -c config.yml
```

Ensure to replace `your_database_connection_string` with your database connection string used in the migration step.

2. Switch traffic to point to the new Ory Enterprise License Hydra. Perform this step on your infrastructure.
3. **Stop the Open Source Version**: Shut down your current open source Ory Hydra service.

```

This guide outlines the steps to upgrade from Ory Hydra open source to Ory Hydra Enterprise License (OEL). Upgrading to OEL
provides additional features and support for enterprise customers.

## Prerequisites

Before starting the upgrade process, ensure you meet the following requirements:

1. **Valid Ory Enterprise License**: You must have purchased an Ory Enterprise License.
2. **Access to the Ory Enterprise Docker Registry**: You need access to download the OEL Docker images.
3. **Ory Hydra Version**: Your current Ory Hydra installation must be version 2.0 or higher.
4. **Backup and Testing**: Create a backup of your current Ory Hydra database and test the migration on a test environment to
   ensure compatibility and minimize risks.

## Upgrade process

Follow these steps to upgrade from Ory Hydra open source to Ory Hydra Enterprise License (OEL):

### 1. Keep the open source version running

Ensure that your current open source version of Ory Hydra is running.

### 2. Run the OEL Image for database migration

Use the Ory Hydra Enterprise License Docker image to run the `hydra migrate sql up` command against the database of the open

> source service. This step applies the necessary database migrations for OEL.

Reviewer: ```suggestion
source service. This step applies the necessary database migrations for OEL. Your current configuration file is compatible with OEL Hydra therefore no changes are required in the file.
```

```bash
docker run --rm \
  -e DSN=your_database_connection_string \
  europe-docker.pkg.dev/ory-artifacts/ory-enterprise/hydra-oel:<version> \
  migrate sql up -e -f config.yml
```

Replace `your_database_connection_string` with your actual database connection string.

### 3. Switch to Ory Hydra Enterprise License

> After successfully migrating the database:

Reviewer: ```suggestion
```
